packages/neptune-extension/src/nbextension/utils/notebook.ts

```diff
@@ -1,34 +1,31 @@
-/* global IPython:readonly */
-
 import { get } from 'lodash';
+import { PlatformNotebookMetadata } from 'types/platform';
 
 export function createPlatformNotebook() {
 
-  function getContent() {
-    return Promise.resolve(JSON.stringify(IPython.notebook.toJSON()));
+  function getContent(): Promise<string> {
+    return Promise.resolve(JSON.stringify(Jupyter.notebook.toJSON()));
   }
 
-  function getMetadata() {
-    const notebook = IPython.notebook;
+  function getMetadata(): PlatformNotebookMetadata {
+    const notebook = Jupyter.notebook;
     const {
-      notebook_name,
       notebook_path,
     } = notebook;
 
     const notebookId = get(notebook, 'metadata.neptune.notebookId');
 
     return {
       path: notebook_path,
-      name: notebook_name,
       notebookId,
     };
   }
 
-  function saveNotebookId(notebookId) {
-    IPython.notebook.metadata.neptune = {
+  function saveNotebookId(notebookId: string) {
```

Comment:
Mark the function as async.
You won't have to write `return Promise.resolve()` because async functions return at least `Promise<void>` anyway.

Reply:
Done